thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MetricBreakdownPipeline.java

```diff
@@ -0,0 +1,242 @@
+package com.linkedin.thirdeye.rootcause.impl;
+
+import com.google.common.collect.Multimap;
+import com.google.common.collect.TreeMultimap;
+import com.linkedin.thirdeye.dataframe.DataFrame;
+import com.linkedin.thirdeye.dataframe.DoubleSeries;
+import com.linkedin.thirdeye.dataframe.StringSeries;
+import com.linkedin.thirdeye.dataframe.util.DataFrameUtils;
+import com.linkedin.thirdeye.dataframe.util.MetricSlice;
+import com.linkedin.thirdeye.dataframe.util.RequestContainer;
+import com.linkedin.thirdeye.datalayer.bao.DatasetConfigManager;
+import com.linkedin.thirdeye.datalayer.bao.MetricConfigManager;
+import com.linkedin.thirdeye.datalayer.dto.DatasetConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
+import com.linkedin.thirdeye.datasource.DAORegistry;
+import com.linkedin.thirdeye.datasource.ThirdEyeCacheRegistry;
+import com.linkedin.thirdeye.datasource.ThirdEyeResponse;
+import com.linkedin.thirdeye.datasource.cache.QueryCache;
+import com.linkedin.thirdeye.rootcause.MaxScoreSet;
+import com.linkedin.thirdeye.rootcause.Pipeline;
+import com.linkedin.thirdeye.rootcause.PipelineContext;
+import com.linkedin.thirdeye.rootcause.PipelineResult;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.collections.MapUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Pipeline for identifying relevant dimensions by performing
+ * contribution analysis. The pipeline first fetches the Current and Baseline entities and
+ * MetricEntities in the search context. It then maps the metrics to ThirdEye's internal database
+ * and performs contribution analysis.
```

Comment:
Could you elaborate and JavaDoc the method that you perform in the analysis? I am not sure I am following this comment.

Reply:
thanks for looking at this. I've expanded the javadoc.

Reply:
I see you have added the explanation for DimensionAnalysisPipeline, which is much clearer than before. Could you JavaDoc the difference between this class and DimensionAnalysisPipeline. Thanks.

Reply:
ok fixed. even more doc